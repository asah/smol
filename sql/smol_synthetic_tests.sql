SET client_min_messages = warning;
CREATE EXTENSION IF NOT EXISTS smol;

-- Test synthetic copy functions by explicitly calling them from SQL
-- This ensures _PG_init and smol_run_synthetic_tests get covered by gcov

-- Call the synthetic test function to trigger coverage of:
-- - _PG_init() code (via explicit call)
-- - smol_run_synthetic_tests()
-- - smol_copy2, smol_copy16, smol_copy_small with various sizes
-- - smol_options() test
SELECT smol_test_run_synthetic();
