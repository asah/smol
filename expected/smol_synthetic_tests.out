-- Test synthetic copy functions by explicitly calling them from SQL
-- This ensures _PG_init and smol_run_synthetic_tests get covered by gcov
CREATE EXTENSION IF NOT EXISTS smol;
NOTICE:  extension "smol" already exists, skipping
-- Call the synthetic test function to trigger coverage of:
-- - _PG_init() code (via explicit call)
-- - smol_run_synthetic_tests()
-- - smol_copy2, smol_copy16, smol_copy_small with various sizes
-- - smol_options() test
SELECT smol_test_run_synthetic();
 smol_test_run_synthetic 
-------------------------
 t
(1 row)

